lib/dns.js

```diff
@@ -125,6 +125,11 @@ exports.lookup = function lookup(hostname, options, callback) {
         hints !== (exports.ADDRCONFIG | exports.V4MAPPED)) {
       throw new TypeError('invalid argument: hints must use valid flags');
     }
+
+    // FIXME(indutny): V4MAPPED on FreeBSD results in EAI_BADFLAGS, because
+    // the libc does not support it
+    if (process.platform === 'freebsd' && family !== 6)
```

Comment:
since the flag is simply invalid, why adding `&& family !== 6` ?

Another (more elaborate) question is whether mapped/not-mapped IPv4 addresses should just have been abstracted from the clients.